src/zjs_gfx.c

```diff
@@ -96,44 +96,60 @@ static void args_to_data(gfx_data_t *data, u32_t argc, const jerry_value_t argv[
 
 static void zjs_gfx_reset_touched_pixels(gfx_handle_t *gfxHandle)
 {
-    gfxHandle->tpX0 = gfxHandle->screenW;
+    gfxHandle->tpX0 = 0;
     gfxHandle->tpX1 = 0;
-    gfxHandle->tpY0 = gfxHandle->screenH;
+    gfxHandle->tpY0 = 0;
     gfxHandle->tpY1 = 0;
     gfxHandle->touched = false;
 }
 
-static void zjs_gfx_touch_pixels(u32_t x, u32_t y, u32_t w, u32_t h, u8_t color[], gfx_handle_t *gfxHandle)
+static void zjs_gfx_touch_pixels(int32_t x, int32_t y, u32_t w, u32_t h, u8_t color[], gfx_handle_t *gfxHandle)
 {
     // Check that x and y aren't past the screen
-    if (x >= gfxHandle->screenW || y >= gfxHandle->screenH)
+    if (x >= gfxHandle->screenW || y >= gfxHandle->screenH || (int32_t)(x + w) <= 0 || (int32_t)(y + h) <= 0) {
+        zjs_gfx_reset_touched_pixels(gfxHandle);
```

Comment:
We shouldn't reset the touched pixels if something is drawn off screen.  Take for example if we are drawing in flush mode, and we have several things being drawn.  Just because some of them are off screen doesn't mean we shouldn't draw the rest.